pkgs/build-support/rust/build-rust-crate/test/default.nix

```diff
@@ -344,6 +344,21 @@ let
         buildTests = true;
         expectedTestOutputs = [ "test baz_false ... ok" ];
       };
+      buildScriptFeatures = {
+        crateName = "foo";
+        features = [ "some-feature" "crate/another_feature" ];
+        src = symlinkJoin {
+          name = "build-script-and-main";
+          paths = [
+            (mkBin  "src/main.rs")
```

Comment:
Should also test that the environment variable is _not_ seen by either `build.rs` nor `src/main.rs` at compile time (e.g. via `env!()` or `option_env!()`.

(I have tested this and the variable appears to be set _only_ when cargo executes build script).

Reply:
Good point! I'll add that to the PR.

Reply:
I added these suggested tests. Also changed the PR to run the build script with `CARGO_FEATURE_*` declarations in a subshell, since they were visible to `main.rs`.